src/ir/access_path.cc

```diff
@@ -0,0 +1,86 @@
+#include "src/ir/access_path.h"
+
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
+#include "absl/strings/match.h"
+#include "absl/strings/string_view.h"
+#include "absl/strings/str_join.h"
+#include "absl/strings/str_cat.h"
+
+#include <string>
+
+namespace raksha::ir {
+
+// Check the components of the two access paths for equality. This is the
+// same as their component vectors being equal according to std::vector's
+// operator==.
+bool AccessPath::operator==(const AccessPath &other) const {
+  return reverse_components_ == other.reverse_components_;
```

Comment:
You shouldn't even need this if all the fields of the class has a valid `==`

Reply:
Are you sure? I think auto-generated C++ `operator==` is a C++20 thing. If I comment out my explicit implementation of `operator==`, my tests fail with compilation errors.